be/src/storage/rowset/beta_rowset_writer.cpp

```diff
@@ -166,12 +168,20 @@ Status BetaRowsetWriter::flush_src_rssids(uint32_t segment_id) {
 }
 
 HorizontalBetaRowsetWriter::HorizontalBetaRowsetWriter(const RowsetWriterContext& context)
-        : BetaRowsetWriter(context), _segment_writer(nullptr) {}
+        : BetaRowsetWriter(context), _segment_writer(nullptr), _final_merge_start_write(false) {}
 
 HorizontalBetaRowsetWriter::~HorizontalBetaRowsetWriter() {
     // TODO(lingbin): Should wrapper exception logic, no need to know file ops directly.
-    if (!_already_built) {       // abnormal exit, remove all files generated
-        _segment_writer.reset(); // ensure all files are closed
+    if (!_already_built) { // abnormal exit, remove all files generated
+        if (_segment_writer != nullptr) {
+            // close the latest write file
+            _segment_writer->close();
```

Comment:
The default operation of close() does not sync the directory.
It seems a bug if we create a new file.